patches/minecraft/net/minecraft/network/NetHandlerPlayServer.java.patch

```diff
@@ -34,6 +34,15 @@
                  this.field_147367_d.func_184103_al().func_148544_a(itextcomponent, false);
              }
  
+@@ -1049,7 +1056,7 @@
+         if (entity != null)
+         {
+             boolean flag = this.field_147369_b.func_70685_l(entity);
+-            double d0 = 36.0D;
++            double d0 = Math.pow(net.minecraftforge.common.ForgeHooks.getPlayerReachDistance(this.field_147369_b) + this.field_147369_b.func_70047_e() + (float)Math.max(entity.func_174813_aQ().field_72336_d - entity.func_174813_aQ().field_72340_a, Math.max(entity.func_174813_aQ().field_72337_e - entity.func_174813_aQ().field_72338_b, entity.func_174813_aQ().field_72334_f - entity.func_174813_aQ().field_72339_c)), 2);
```

Comment:
where is this logic coming from?

Reply:
> To fix this, I made the check's distance the maximum it could be by adding the player's eye height and the biggest side of the entity's bounding box. If you can think of a better solution, let me know.

^ It's the fix I mentioned in the PR. 

Can you think of a better way to do that? I'm open to suggestions

Reply:
Replace the call to getDistanceSq() with a new Forge hook (rendering the d0 variable unusable):
```java
    public static boolean getPlayerCanReach(EntityPlayerMP player, Entity entity) {
        Vec3d eyePos = player.getPositionEyes(1.0F);
        AxisAlignedBB bBox = entity.getEntityBoundingBox();
        if (bBox.contains(eyePos)) return true;
        double dx = 0.0D, dy = 0.0D, dz = 0.0D;
        if (!(bBox.minX <= eyePos.x && eyePos.x <= bBox.maxX))
            dx = Math.min(Math.pow(eyePos.x - bBox.minX, 2), Math.pow(eyePos.x - bBox.maxX, 2));
        if (!(bBox.minY <= eyePos.y && eyePos.y <= bBox.maxY))
            dy = Math.min(Math.pow(eyePos.y - bBox.minY, 2), Math.pow(eyePos.y - bBox.maxY, 2));
        if (!(bBox.minZ <= eyePos.z && eyePos.z <= bBox.maxZ))
            dz = Math.min(Math.pow(eyePos.z - bBox.minZ, 2), Math.pow(eyePos.z - bBox.maxZ, 2));
        return Math.pow(getPlayerReachDistance(player), 2) >= dx + dy + dz;
    }```
This function gets player's reach distance and eye position, calculates minimum distance from that position to the entity's hitbox and compares it to reach distance. If that distance is less than reach distance - the player can reach that entity.